### PR TITLE
[GradleV3] Improve task failure message

### PR DESCRIPTION
**Pull request by alexander-smolyakov**
**Task name**: GradleV3

**Description**:
Improve task failure message to make it user-friendly

_Changelog_:
- Improved task failure message
- Moved helper functions to separate modules
- Added interfaces
- Resolved linter warnings
- Updated linter settings 
- Added JSdoc to the functions

**Documentation changes required:** No

**Added unit tests:** No

**Attached related issue:**
- #3472

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/GradleV3/Modules/utils.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    if (codeAnalysisResult.gradleResult === 0) {
+        status = TaskResult.Succeeded;
+        message = 'Build succeeded.';
+    } else if (codeAnalysisResult.gradleResult === -1 && codeAnalysisResult.statusFailed === true) {
+        status = TaskResult.Failed;
+        message = codeAnalysisResult.analysisError;
+    } else {
+        status = TaskResult.Failed;
+        message = 'Build failed.';
```
Comment (anatolybolshakov) [inline Tasks/GradleV3/Modules/utils.ts:21]: Could it make sense to write more info here? Probably just all codeAnalysisResult fields?

File: Tasks/GradleV3/gradletask.ts [hunk 2]
```diff
@@ -221,96 +51,111 @@ async function run() {
...
+        const publishCodeCoverageSettings: IPublishCodeCoverageSettings = {
+            isCodeCoverageOpted: isCodeCoverageOpted,
+            failIfCoverageEmpty: failIfCodeCoverageEmpty,
+            codeCoverageTool: codeCoverageTool,
+            summaryFile: summaryFile,
+            reportDirectory: reportDirectory
+        };
+        await publishCodeCoverageResultsAsync(publishCodeCoverageSettings);
+
+        const taskResult: ITaskResult = resolveTaskResult(codeAnalysisResult);
```
Comment (anatolybolshakov) [inline Tasks/GradleV3/gradletask.ts:157]: Could it make sense to add input to be able to turn off this logic? Is this backward compatible?

File: Tasks/GradleV3/Modules/project-configuration.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as tl from 'azure-pipelines-task-lib/task';
+import { ToolRunner } from 'azure-pipelines-task-lib/toolrunner';
+
+/**
+ * Check to determine if if project multi module or not
```
Comment (EzzhevNikita) [inline Tasks/GradleV3/Modules/project-configuration.ts:7]: nit: extra `if`

**Review by EzzhevNikita — APPROVED** [review]

File: Tasks/GradleV3/Modules/project-configuration.ts (hunk 3 above)
Comment (alexander-smolyakov) [reply Tasks/GradleV3/Modules/project-configuration.ts]: Nice catch, fixed it.

File: Tasks/GradleV3/Modules/utils.ts (hunk 1 above)
Comment (alexander-smolyakov) [reply Tasks/GradleV3/Modules/utils.ts]: Updated the error message.

File: Tasks/GradleV3/gradletask.ts (hunk 2 above)
Comment (alexander-smolyakov) [reply Tasks/GradleV3/gradletask.ts:157]: I double-checked my changes and I believe that we don't have to hide this functionality behind the task input, the changes themselves are backward compatible.

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

**Review by KonstantinTyukalov — APPROVED** [review]
I checked these changes manually. LGTM, Thanks!

**Comment by alexander-smolyakov** [comment]
Hey @mmrazik, could you please review and merge this pull request?

**Review by mmrazik — APPROVED** [review]

**Pull request merged**